internal/pkg/manifest/backend_svc.go

```diff
@@ -188,3 +189,26 @@ func newDefaultBackendService() *BackendService {
 		Environments: map[string]*BackendServiceConfig{},
 	}
 }
+
+// ExposedPorts returns all the ports that are container ports available to receive traffic.
+func (b *BackendService) ExposedPorts() []ExposedPort {
+
+	var exposedPortList []ExposedPort
+	exposedPorts := make(map[int]ExposedPort)
+
+	workloadName := aws.StringValue(b.Name)
+	b.ImageConfig.exposedPorts(exposedPorts, workloadName)
+	b.RoutingRule.exposedPorts(exposedPorts, workloadName)
```

Comment:
maybe these are from a different PR, but how are we handling conflicts? shouldn't that be an error when parsing the manifest? 

if we already know the port numbers are unique (i.e., errors occurred before this is called), then i'd prefer to make this return a slice instead of a map so we can just have a single `exposedPorts` slice instead of a map+slice

Reply:
looks like validate will have already run (i'm assuming) - so maybe we can drop the map and just create a single slice?

Reply:
Yeah validation will take care of the duplicates, so probably I can remove map from here and just use slice.